libparsec/crates/platform_storage/src/web/model.rs

```diff
@@ -0,0 +1,195 @@
+
+use crate::PREVENT_SYNC_PATTERN_EMPTY_PATTERN;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(super) struct Certificate {
+    pub certificate_timestamp: Float,
+    pub certificate: Bytes,
+    pub certificate_type: String,
+    pub filter1: String,
+    pub filter2: String,
+}
+
+impl Certificate {
+    const STORE: &'static str = "certificates";
+    const INDEX_CERTIFICATE_TYPE: &'static str = "certificate_type";
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(super) struct Chunk {
+    pub chunk_id: Bytes,
+    pub size: IndexInt,
+    pub offline: bool,
+    pub accessed_on: Float,
+    pub data: Bytes,
+}
+
+impl Chunk {
+    const STORE: &'static str = "chunks";
+    const INDEX_CHUNK_ID: &'static str = "chunk_id";
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(super) struct PreventSyncPattern {
+    pub id: IndexInt,
+    pub pattern: String,
+    pub fully_applied: bool,
+}
+
+impl PreventSyncPattern {
+    const STORE: &'static str = "prevent_sync_pattern";
+    const INDEX_ID: &'static str = "id";
+
+    pub(super) async fn insert_if_empty(self, conn: &IdbDatabase) -> anyhow::Result<()> {
+        super::db::insert_if_empty(conn, Self::STORE, self).await
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(super) struct RealmCheckpoint {
+    pub id: IndexInt,
+    pub checkpoint: IndexInt,
+}
+
+impl RealmCheckpoint {
+    const STORE: &'static str = "realm_checkpoint";
+    const INDEX_ID: &'static str = "id";
+
+    pub(super) async fn get(conn: &IdbDatabase, id: IndexInt) -> anyhow::Result<Option<Self>> {
+        super::db::get_value(conn, Self::STORE, Self::INDEX_ID, id).await
+    }
+
+    pub(super) async fn insert(self, conn: &IdbDatabase) -> anyhow::Result<()> {
+        super::db::insert(conn, Self::STORE, self).await
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(super) struct Remanence {
+    pub id: IndexInt,
+    pub block_remanent: bool,
+}
+
+impl Remanence {
+    const STORE: &'static str = "remanence";
+    const INDEX_ID: &'static str = "id";
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub(super) struct Vlob {
+    pub vlob_id: Bytes,
+    pub base_version: VersionInt,
+    pub remote_version: VersionInt,
+    pub need_sync: bool,
+    pub blob: Bytes,
+}
+
+impl Vlob {
+    const STORE: &'static str = "vlobs";
+    const INDEX_VLOB_ID: &'static str = "vlob_id";
+
+    pub(super) async fn get(conn: &IdbDatabase, vlob_id: Bytes) -> anyhow::Result<Option<Self>> {
+        super::db::get_value(conn, Self::STORE, Self::INDEX_VLOB_ID, vlob_id).await
+    }
+
+    pub(super) async fn insert(self, conn: &IdbDatabase) -> anyhow::Result<()> {
+        super::db::insert(conn, Self::STORE, self).await
+    }
+}
+
+pub(super) async fn initialize_model_if_needed(
+    mut db_req: OpenDbRequest,
+) -> anyhow::Result<IdbDatabase> {
+    db_req.set_on_upgrade_needed(Some(|evt: &IdbVersionChangeEvent| {
+        if let None = evt.db().object_store_names().find(|n| n == Vlob::STORE) {
```

Comment:
```suggestion
        if evt.db().object_store_names().find(|n| n == Vlob::STORE).is_none() {
```

We don't use the value so you can directly use `.is_none()`

Reply:
yes of course, I took this from the crate example :sweat_smile:

Reply:
Done